app/controllers/offices_controller.rb

```diff
@@ -1,22 +1,41 @@
 class OfficesController < ApplicationController
   def index
+    # return on direct requests to the page
     return unless search
-    return @error_message = search.error_message unless search.valid?
 
-    @transportation_offices = TransportationOffice.by_distance_with_shipping_office(search.result).paginate(page: params[:page])
+    # return error message on invalid search
+    return error_message unless search.valid?
+
+    # return search results on valid search (GET)
+    return transportation_offices unless request.post?
+
+    # redirect on valid search (POST)
+    redirect_to offices_path(search.result)
```

Comment:
one UI change this introduces is now zip code searches (and also installation searches although those are new), will say "Displaying results near <lat>,<long>" instead of "Displaying results near <zip code>". It'd be nice to preserve the original query since that's kinda confusing